include/sensors/firmware/sensor_hardware.hpp

```diff
@@ -5,12 +5,22 @@
 
 namespace sensors {
 namespace hardware {
+
+struct SensorHardwareConfiguration {
+    gpio::PinConfig sync_in;
+    gpio::PinConfig sync_out;
+    gpio::PinConfig data_ready;
+};
+
 class SensorHardware : public SensorHardwareBase {
   public:
-    explicit SensorHardware(gpio::PinConfig sync) : sync_pin(sync) {}
-    auto set_sync() -> void override { gpio::set(sync_pin); }
-    auto reset_sync() -> void override { gpio::reset(sync_pin); }
-    gpio::PinConfig sync_pin;
+    SensorHardware(SensorHardwareConfiguration hardware) : hardware(hardware) {}
+    auto set_sync_in() -> void override { gpio::set(hardware.sync_in); }
```

Comment:
Yeah these absolutely need to be `sync_out` not `sync_in` - `sync_in` is the one you read, and `sync_out` is the one you write